src/main/java/net/minecraftforge/fml/common/FMLCommonHandler.java

```diff
@@ -635,8 +636,9 @@ public boolean handleServerHandshake(C00Handshake packet, NetworkManager manager
         if (packet.getRequestedState() == EnumConnectionState.LOGIN && (!NetworkRegistry.INSTANCE.isVanillaAccepted(Side.CLIENT) && !packet.hasFMLMarker()))
         {
             manager.setConnectionState(EnumConnectionState.LOGIN);
-            TextComponentString text = new TextComponentString("This server requires FML/Forge to be installed. Contact your server admin for more details.");
-            FMLLog.log.info("Disconnecting Player: {}", text.getUnformattedText());
+            TextComponentString text = new TextComponentString("This server has mods that require FML/Forge to be installed on the client. Contact your server admin for more details.");
+            Collection<String> modNames = NetworkRegistry.INSTANCE.getModNamesThatDoNotAcceptVanilla(Side.CLIENT);
+            FMLLog.log.info("Disconnecting Player: This server has mods that require FML/Forge to be installed on the client:\n{}", modNames);
```

Comment:
I'd put a tab after the new line to indicate the messages go together. But looks fine to me. Except your method name is pretty big, could just be 'getRequiredMods(SIDE)'

Reply:
Sounds good, will do.

Reply:
I'll just put it on one line, it's going to be really long either way and this way it'll all be together.